src/main/daml/Daml/Finance/Common/Date/Schedule.daml

```diff
@@ -83,9 +85,12 @@ createSchedule : [HolidayCalendarData] -> PeriodicSchedule -> Schedule
 createSchedule _ ps | Some errorMsg <- validDates ps = error errorMsg
 createSchedule cals ps@PeriodicSchedule{businessDayAdjustment; effectiveDateBusinessDayAdjustment; terminationDateBusinessDayAdjustment} =
   let
-    unadj = generateUnadjustedDates ps
+    (unadj, stubTypeInitial, stubTypeFinal) = generateUnadjustedDates ps
+    stubTypesDefault = (replicate (length unadj - 1) None)
+    stubTypesInclInitial = if (isSome stubTypeInitial) then stubTypeInitial :: (tail stubTypesDefault) else stubTypesDefault
+    stubTypes = if (isSome stubTypeFinal) then (init stubTypesInclInitial) ++ [stubTypeFinal] else stubTypesInclInitial
```

Comment:
I would move this list creation logic inside the `toPeriod` function. This makes it easier to understand why we build a list of length `len unadj - 1`.

Also, you could build it as `stubTypes = [stubTypeInitial] ++  replicate (length unadj - 3) None ++ [stubTypeFinal]`

Reply:
Actually, my suggestion might not work in the case of a 1-period schedule... not sure how we want to handle that case, would you know if there is a test for it?

Reply:
I feel that a 1-period schedule doesn't really have a notion of a stub.

Reply:
> I would move this list creation logic inside the `toPeriod` function. This makes it easier to understand why we build a list of length `len unadj - 1`.

Done

Reply:
> Also, you could build it as `stubTypes = [stubTypeInitial] ++ replicate (length unadj - 3) None ++ [stubTypeFinal]`

I started out with this solution, but as you point out, I thought it might not handle 1 period schedules well.

Reply:
> Actually, my suggestion might not work in the case of a 1-period schedule... not sure how we want to handle that case, would you know if there is a test for it?

Yes, there are a few of them already. Useful to make sure that the date creation works as expected, even though the stub info does not make much sense as you point out.